docs/cpan-dependency-confusion.md

```diff
@@ -0,0 +1,238 @@
+# CPAN dependency confusion
+## Rationale 
+Take advantage of a wrongly configured setup that resolve to public module in place of expected private/local version, because of one of the following reasons:
+- Local module is not installed at time of running CPAN client
+- Public CPAN version is preferred to local
+- Using a CPAN repository (proxy) that can be confused
+
+This type of vulnerability is presented in [Dependency Confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610) by [Alex Birsan](https://medium.com/@alex.birsan)
+
+## Background
+"Dependency confusion" takes advantage of mixing (semi-)private modules with public CPAN modules.
+
+In Perl ecosystem, it could be one or more of the following options:
+- Using a "DarkPAN" along with CPAN
+- Using a "GreyPAN" along with CPAN
+- Mixing manual installs (`make install`, "blib install"...) along with CPAN
+
+See [About the various PANs](https://www.olafalders.com/2019/02/19/about-the-various-pans/) for details about DarkPAN and GreyPAN.
+
+An attacker can "claim" a public namespace, install a high version (to take precedence in index over a smaller local version) and wait for vulnerable setups to install (and execute) their newly uploaded module.
+
+## General about CPAN dependency management
+### The mirror versus the indexer
+The notions of "mirror" and "indexer":
+- "mirror" - where module tarballs are stored
+- "indexer" - translate module to path and version
+
+These notions are not always well differenciated and their meaning heavily depends the CPAN client. For instance, from a pool (urllist), `CPAN.pm` will use only one mirror for both index and download of modules. On the other hand, `cpanm` will use MetaDB as index in priority and mirrors to download, and these notions are well decoupled. 
+
```

Comment:
How does BackPAN (an archive of deleted versions of modules) related to this?

Carton will use that, for instance.

Reply:
Question: are modules found to be "malicious" or "typo squatting" also deleted from BackPAN?

Reply:
I need to do some more research to answer these 2 questions

Reply:
It's interesting to know how much malicious packages can survive from deletion

Reply:
Did my comment about the Perl modules list for August 2018 get zapped?

See https://www.nntp.perl.org/group/perl.modules/2018/08.html for an example of spam/malicious authors and modules.

Reply:
I think at some point, there's no much distinction between CPAN and BackPAN.

If you look at what happens when requesting a version that is on BackPAN, it's transparent for us:
```
$ cpanm -L local Moose@2.2206
--> Working on Moose
Fetching http://backpan.perl.org/authors/id/E/ET/ETHER/Moose-2.2206.tar.gz ... OK
```